### PR TITLE
Add indented nav-pills navigation option for ToC

### PR DESCRIPTION
**Pull request by jason-fox**
Additional ToC options using **indented nav-pills** - list-groups aren't really so useful for deep hierarchies.

- Updated available parameters
- Updated XSL
- Updated Documentation
- Added new images to the carousel.

`list-group` has been altered to use `bg-light` so the CSS override is no longer necessary.

### Nav-pill full

nav-pill-full

### Nav-pill partial

nav-pill-partial

### Addition of `menubar-toc.include` parameter

Adds a header submenu with dropdowns as the first layer of navigation. Reduces the sidebar to active elements only. This can eliminate the need for sidebar altogether in some cases. The existing sample works well as `--nav-toc=none --menubar-toc-include=yes' for example`

Screenshot 2022-01-09 at 16 31 36

Screenshot 2022-01-09 at 16 28 18

### CODE REVIEW

**Comment by jason-fox** [comment]
### Scrollspy Navigation Option
Automatically update a side-bar ToC using Bootstrap navigation or list group components based on scroll position to indicate which link is currently active in the viewport. Unlike other ToC options, scrollspy works only on embedded sub-topics within DITA files. The sidebar will not contain a topic tree based in the main ditamap.

#### `nav-toc=nav-pill-scrollspy`

nav-pill-scrollspy

#### `nav-toc=list-group-scrollspy`

list-group-scrollspy

**Comment by jason-fox** [comment]
### Skip to Main Content Link - Hidden by default.

accessibility

Additional hidden keyboard navigation links for accessibility - see https://www.youtube.com/watch?v=jDDaOFr9nqQ

File: Customization/xsl/carousel.xsl [hunk 1]
```diff
@@ -29,12 +29,20 @@
...
+          </xsl:call-template>
+        </span>
       </a>
       <a class="carousel-control-next" role="button" data-bs-slide="next">
         <xsl:attribute name="data-bs-target" select="concat('#' , $id)"/>
         <span class="carousel-control-next-icon" aria-hidden="true"/>
-        <span class="visually-hidden">Next</span>
+        <span class="visually-hidden">
+          <xsl:call-template name="getVariable">
+            <xsl:with-param name="id" select="'Previous'"/>
```
Comment (infotexture) [inline Customization/xsl/carousel.xsl:43]: Should this be **Next**?

```suggestion
            <xsl:with-param name="id" select="'Next'"/>
```

File: sample/scrollspy.dita [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<!-- Within the sample documentation, where necessary, the texts describing the
+   usage of each component have been copied directly from the official Bootstrap
+   5.0 documentation (found at https://getbootstrap.com/docs/5.0), however DITA
```
Comment (infotexture) [inline sample/scrollspy.dita:5]:
```suggestion
   5.1 documentation (found at https://getbootstrap.com/docs/5.1), however DITA
```

**Review by infotexture — APPROVED** [review]
Sorry it took so long to review this, but there was so much more going on here than just the indented nav pills, so I kept putting it off until I had the time to test properly.

That said, the new things I see here all make sense, including: 

- CSS grid for layout
- Scrollspy for nested topics
- New accessibility navigation
- New menubar ToC options
- Adding localized strings for "skip to content" navigation
- Renaming common Bootstrap CSS & splitting out ToC CSS variants

In the future, if we manage to keep PRs more focused, I should be able to review faster.

There are a few things here I think we'll want to tweak further before releasing, but we can treat those as separate issues rather than holding this up any further:

- [ ] Remove `active` class from active page ancestors
  _(only one page is really "active")_
- [ ] Menu bar ToC should respect parent container
  Currently, if `--menubar-toc.include=yes`, the ToC menu is left-aligned, whereas the menu above is centered within the parent container

**Pull request merged**